pentaho-service-coordinator/src/test/java/org/pentaho/platform/servicecoordination/impl/ServiceBarrierThreadTest.java

```diff
@@ -0,0 +1,133 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2002-2015 Pentaho Corporation..  All rights reserved.
+ */
+package org.pentaho.platform.servicecoordination.impl;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.pentaho.platform.servicecoordination.api.IServiceBarrier;
+
+/**
+ * 
+ * @author tkafalas
+ *
+ */
+public class ServiceBarrierThreadTest {
+  final static String BARRIER_ID = "JCRBarrier";
+
+  /**
+   * Test simulates 3 features waiting for the main thread to release the barrier before continuing.
+   * 
+   * @throws Exception
+   */
+  @Test
+  public void testMultipleThreads1() throws Exception {
+
+    final ServiceBarrierManager mngr = new ServiceBarrierManager();
+    IServiceBarrier barrier = mngr.getServiceBarrier( BARRIER_ID );
+    assertTrue( barrier.isAvailable() );
+
+    // barrier.hold();
+    barrier.hold();
+
+    Thread t1 = new Thread( new ServiceTestThread( mngr, 1 ) );
+    Thread t2 = new Thread( new ServiceTestThread( mngr, 2 ) );
+    Thread t3 = new Thread( new ServiceTestThread( mngr, 3 ) );
+    t1.start();
+    t2.start();
+    t3.start();
+
+    Thread.sleep( 3000 );
```

Comment:
3 seconds is a long wait for a test. It's better to loop through them all until they're waiting.